packages/gatsby/src/datastore/lmdb/query/__tests__/suggest-index.ts

```diff
@@ -0,0 +1,313 @@
+
+        // TODO: consider sorting "eq" fields in the order they appear in "sort"
+        //  The assumption here is that the same sort order may be used somewhere else,
+        //  so maybe this index can be re-used more often
+
+        expect(suggestIndex({ filter, sort })).toEqual([
+          [`a`, 1],
+          [`c`, 1],
+          [`b`, 1],
+        ])
+      })
+
+      it(`removes "eq" filters from "sort" tail`, () => {
+        const filter = { b: { eq: `foo` }, c: { eq: `bar` } }
+        const sort: Sort = {
+          fields: [`a`, `b`, `c`],
+          order: [`desc`, `desc`, `desc`],
+        }
+        expect(suggestIndex({ filter, sort })).toEqual([
+          [`b`, -1],
+          [`c`, -1],
+          [`a`, -1],
+        ])
+      })
+
+      it(`removes "eq" fields from the middle of "sort" list`, () => {
+        const filter = { b: { eq: `bar` }, c: { eq: `bar` } }
+        const sort: Sort = { fields: [`a`, `c`, `b`, `d`], order: [] }
+
+        expect(suggestIndex({ filter, sort })).toEqual([
+          [`b`, 1],
+          [`c`, 1],
+          [`a`, 1],
+          [`d`, 1],
+        ])
+      })
+    })
+
+    describe(`single "in" filter`, () => {
+      it(`prefers "in" filter to "sort" fields`, () => {
+        // TODO: maybe it's actually possible to support this scenario.
+        //  As each entry of "in" array runs a separate range query anyway.
+        //  So
+        //  { a: 1, b: 100, id: 1 }, { a: 1, b: 500, id: 2 } and
+        //  { a: 2, b: 1, id: 3 }, { a: 2, b: 600, id: 4 }
+        //  can potentially exploit the fact that they are sorted within each bucket and use mergeSorted to
+        //  traverse in order: 3, 1, 2, 4
+        //  it doesn't require deduplication (unless it is a MultiKey index)
+        //  Can be costly if "in" has too many values (but probably not costlier than in-memory sort anyway)
+        const filter = { a: { in: [1, 2] } }
+        const sort = { fields: [`b`, `c`], order: [] }
+        expect(suggestIndex({ filter, sort })).toEqual([[`a`, 1]]) // TODO: .toEqual([[`a`, 1], [`b`, 1], [`c`, 1]]])
+      })
+
+      it(`merges "in" filter with sibling "sort" field`, () => {
+        const filter = { a: { in: `foo` } }
+        const sort: Sort = { fields: [`a`, `b`], order: [`desc`, `desc`] }
+        expect(suggestIndex({ filter, sort })).toEqual([
+          [`a`, -1],
+          [`b`, -1],
+        ])
+      })
+
+      it(`discards "sort" fields that do not overlap with "in" filter`, () => {
+        const filter = { b: { in: [`foo`] } }
+        const sort: Sort = { fields: [`a`, `b`, `c`], order: [] }
+        expect(suggestIndex({ filter, sort })).toEqual([[`b`, 1]])
+      })
+    })
+
+    describe.each([`lt`, `lte`, `gt`, `gte`])(
+      `single "%s" filter`,
+      predicate => {
+        it(`prefers "sort" fields to single "${predicate}" filter`, () => {
+          const filter = { a: { [predicate]: [1, 2] } }
```

Comment:
is `[1, 2]` proper value for `lt` etc comparators? does it even matter for this test?

Reply:
Addressed in https://github.com/gatsbyjs/gatsby/pull/32261/commits/b6a8c581b768cc49aa843216cccba1eee999859f

Reply:
Addressed in https://github.com/gatsbyjs/gatsby/pull/32261/commits/b6a8c581b768cc49aa843216cccba1eee999859f